### PR TITLE
Dan/apps view e2e dev

### PR DESCRIPTION
**Pull request by danielsdeleo**
### :nut_and_bolt: Description

Very minimal "dev environment" for running a real hab-sup against automate running in the studio.

### :white_check_mark: Checklist

- [ ] Necessary tests added/updated?
- [ ] Necessary docs added/updated?
- [ ] Code actually executed?
- [ ] Vetting performed (unit tests, lint, etc.)?

### CODE REVIEW

File: components/applications-service/dev/hab-client/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,22 @@
+FROM ubuntu:18.04
+MAINTAINER Chef Software <dan@chef.io>
```
Comment (afiune) [inline components/applications-service/dev/hab-client/Dockerfile]: Do we want Dan to be the maintainer of this docker image? 🤔

File: components/applications-service/dev/hab-client/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+
+ENV HAB_LICENSE=accept
+
+RUN curl https://raw.githubusercontent.com/habitat-sh/habitat/master/components/hab/install.sh | bash -s -- -c unstable && \
+    hab pkg install core/hab-sup/0.83.0-dev -c unstable && \
+    apt-get clean && rm -rf /var/lib/apt/lists/* /tmp/* /var/tmp/*
+
+ENV PATH=/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin
+
+#CMD ["bash", "-c", "exec /bin/hab sup run"]
```
Comment (afiune) [inline components/applications-service/dev/hab-client/Dockerfile]: Do we need this command or should we remove the comment?

File: components/applications-service/dev/hab-client/README.md [hunk 3]
```diff
@@ -0,0 +1,9 @@
+# Hab Client End to End Docker Environment
+
+## Short Version
+
+1. If you don't have Automate running in the studio, enter the studio and
+   `start_all_services`
+2. In the studio, get an API token with `get_api_token`
+3. Run the script: `./run-hab-sup.sh -t YOUR_TOKEN`
```
Comment (afiune) [inline components/applications-service/dev/hab-client/README.md]: Do we run this inside the studio? so means that we will start a new docker container inside the docker container? 🚢 ⛵️

File: components/applications-service/dev/hab-client/run-hab-sup.sh [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+  # Downside of this approach is we are not being smart about the caching and
+  # folks will have to know when/how to clean the cache and force a new image
+  # build.
+  # FYI, the way to nuke the image:
+  # docker rmi -f chef/automate-apps-hab-client
+  docker build . -t chef/automate-apps-hab-client
+}
+
+run_hab_in_container() {
+  docker run -it chef/automate-apps-hab-client bash -c "
```
Comment (afiune) [inline components/applications-service/dev/hab-client/run-hab-sup.sh]: Is this command being run inside the studio or outside? 🤔

**Comment by danielsdeleo** [comment]
@afiune below is the README as it is now, is it clear enough?

# Hab Client End to End Docker Environment

## Short Version

1. If you don't have Automate running in the studio, enter the studio and
   `start_all_services`
2. In the studio, get an API token with `get_api_token`
3. Until the habitat supervisor supports NATS TLS, you need to turn off TLS on
   the automate event-gateway. Use the following toml and enable it with
   `chef-automate config patch` (not setting this will cause an error like
   `Could not establish streaming connection to NATS server -> timed out
   waiting on channel`):
   ```
   [event_gateway]
     [event_gateway.v1]
       [event_gateway.v1.sys]
         [event_gateway.v1.sys.service]
           disable_frontend_tls = true
   ```   
4. Outside of the studio, run the script: `./run-hab-sup.sh -t YOUR_TOKEN`

## Description

The Dockerfile in this directory defines a docker image with a real habitat
client that you can connect to your Automate development environment to test
the EAS Applications functionality with a real Habitat supervisor. It's
designed to run on MacOS and uses a Docker for Mac-specific feature.

File: components/applications-service/dev/hab-client/run-hab-sup.sh [hunk 5]
```diff
@@ -0,0 +1,84 @@
...
+
+  # Parse command line flags and options.
+  while getopts ":ht:" opt; do
+    case "${opt}" in
+      h)
+        print_help
+        exit 0
+        ;;
+      t)
+        token="${OPTARG}"
```
Comment (afiune) [inline components/applications-service/dev/hab-client/run-hab-sup.sh]: nit: should we check when the provided token (that is the flag) is empty?

Comment (danielsdeleo) [reply components/applications-service/dev/hab-client/run-hab-sup.sh]: added!

**Pull request merged**